papas/datatypes/src/Cluster.cpp

```diff
@@ -27,19 +27,44 @@ Cluster::Cluster(const Cluster& c, unsigned int index, IdCoder::ItemType type, c
     : m_id(IdCoder::makeId(index, type, subtype, val)),
       m_size(c.m_size),
       m_angularSize(c.m_angularSize),
-      m_pt(c.m_pt),
       m_p3(c.m_p3),
       m_energy(c.m_energy),
       m_subClusters() {
   ;
   m_subClusters.push_back(&c);
 }
 
+Cluster::Cluster(std::list<const Cluster*> overlappingClusters, unsigned int index, char subtype)
+    : m_subClusters(overlappingClusters) {
+  Identifier firstId = -1;
+  char type;
+  for (const auto* cluster : overlappingClusters) {
+    if (cluster->subClusters().size() > 1) {
+      throw "can only merge clusters which are not already merged";
+    }
+    if (firstId == -1) {
+      firstId = cluster->id();
+      m_p3 = cluster->position() * cluster->energy();
```

Comment:
I think there is a bug here. 
m_p3 is now proportionnal to the magnitude of the vector that goes from the origin to the position of the cluster in 3D space